sys/net/gnrc/network_layer/sixlowpan/frag/rbuf.c

```diff
@@ -87,11 +87,16 @@ static int _rbuf_add(gnrc_netif_hdr_t *netif_hdr, gnrc_pktsnip_t *pkt,
                      size_t offset, unsigned page)
 {
     rbuf_t *entry;
-    sixlowpan_frag_t *frag = pkt->data;
+    sixlowpan_frag_n_t *frag = pkt->data;
     rbuf_int_t *ptr;
     uint8_t *data = ((uint8_t *)pkt->data) + sizeof(sixlowpan_frag_t);
     size_t frag_size;
 
+    /* check if provided offset is the same as in fragment */
+    assert(((((frag->disp_size.u8[0] & SIXLOWPAN_FRAG_DISP_MASK) ==
+                SIXLOWPAN_FRAG_1_DISP)) && (offset == 0)) ||
+           ((((frag->disp_size.u8[0] & SIXLOWPAN_FRAG_DISP_MASK) ==
+                SIXLOWPAN_FRAG_N_DISP)) && (offset == (frag->offset * 8U))));
```

Comment:
`gnrc_sixlowpan_frag` internally derives the offset value directly from the fragment header

https://github.com/RIOT-OS/RIOT/blob/1f7292c358ccda95090fb15ff29ac93d674f26e2/sys/net/gnrc/network_layer/sixlowpan/frag/gnrc_sixlowpan_frag.c#L306-L322

So for normal usage within GNRC this assertion is redundant, but to make the tests of `rbuf_add` 100% water-tide I added it in a separate commit (48d73191c5e6161ed1c2474004581b14d5a2062b when writing this comment).